contributors/design-proposals/storage/local-storage-pv.md

```diff
@@ -838,19 +847,12 @@ We'll consider it in a subsequent phase.
 
 ##### Change in Populator
 
-In addition to PV, the `populator` will also be extended to watch StorageClasses and PVC objects of all namespaces.
-
-It stores latest StorageClass objects in cache, they are used:
-
-* To reconcile capacity, the StorageManager need to compare actual capacity with that recorded
-in StorageClass objects
-* During volume creation, some opitions (e.g. fsType for `mkfs`) are specified in StorageClasses as Parameters
-
-And for PVCs, when the `populator` finds:
+In addition to PV, the `populator` will also be extended to watch PVC objects of all namespaces.
+When the `populator` finds:
 
 1. An unbound PVC is annotated with `annSelectedNode`, and its node name match the value
-of the annotation
-2. Value of the `annStorageProvisioner` on the PVC matches the name of the provisioner
+of the annotation.
+2. Value of the `annStorageProvisioner` on the PVC matches the name of the provisioner.

```

Comment:
Per my latest update of the topology design, we should also add 'annProvisionerTopology' to this list of required fields

Reply:
I think once `annSelectedNode` is set, the topology of the node should have been measured by the scheduler.
So here we need to make a double check to prevent the topology label of the node from being changed midway?

Reply:
The check for `annProvisionerTopology` is to handle the coordination between provisioners that need to report capacity and initialization of the StorageClass.  If `annProvisionerTopology` was not set, then that means the scheduler made a decision not taking into account capacity.

Reply:
Can the removal of the annotation be one level higher so that it's common across all provisioner implementations?

Reply:
Yes, what I meant was the "general" storagemanager layer, not the lvm specific layer.

Reply:
sure thing :)